File: PyProject/leetcode/history/minimum-depth-of-binary-tree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class MaximumDepthOfBinaryTree:
    def minDepth(self, root: TreeNode) -> int:
        if not root:
            return 0
        if not root.left:
            return self.minDepth(root.right) + 1
        if not root.right:
            return self.minDepth(root.left) + 1
        return min(self.minDepth(root.right), self.minDepth(root.left)) + 1

File: GoProject/leetcode/dfs/minimum-depth-of-binary-tree.go
package dfs

func minDepth(root *TreeNode) int {
	if root.Right == nil && root.Left == nil {
		return 1
	}

	right := minDepth(root.Right) + 1
	left := minDepth(root.Left) + 1
	if right > left {
		return left
	}
	return right

}
